### PR TITLE
refactor(phrases,shared,connector-core): move language key to shared

### PR DESCRIPTION
**Pull request by wangsijie**
## Summary

Rename type `Language` to `LanguageKey` and move to `shared`, so that connectors can use them directly.

## Testing

UTs

### CODE REVIEW

File: packages/schemas/src/seeds/setting.ts [hunk 1]
```diff
@@ -9,7 +7,7 @@ export const createDefaultSetting = (): Readonly<CreateSetting> =>
   Object.freeze({
     id: defaultSettingId,
     adminConsole: {
-      language: Language.English,
+      language: 'en',
```
Comment (darcyYe) [inline packages/schemas/src/seeds/setting.ts:10]: should also fix the use of `Language` in packages/schemas/src/seeds/sign-in-experience.ts?

Comment (darcyYe) [reply packages/schemas/src/seeds/setting.ts:10]: hence we can remove the dependencies on @logto/phrases(-ui)?

File: packages/console/src/hooks/use-user-preferences.ts [hunk 2]
```diff
@@ -11,7 +11,7 @@ import { themeStorageKey } from '@/consts';
 import useApi, { RequestError } from './use-api';
 
 const userPreferencesGuard = z.object({
-  language: z.nativeEnum(Language).optional(),
+  language: z.enum(languageKeys).optional(),
```
Comment (darcyYe) [inline packages/console/src/hooks/use-user-preferences.ts:14]: maybe use languageKeyGuard.optional() which has already been defined in @logto/shared?

**Review by darcyYe — COMMENTED** [review]
Also maybe we can double check the failure on PR checks to find more unfixed Language use cases.

File: packages/console/src/hooks/use-user-preferences.ts (hunk 2 above)
Comment (wangsijie) [reply packages/console/src/hooks/use-user-preferences.ts:14]: It will failed

File: packages/schemas/src/seeds/setting.ts (hunk 1 above)
Comment (wangsijie) [reply packages/schemas/src/seeds/setting.ts:10]: It is already done?

File: packages/connector-core/src/types.ts [hunk 3]
```diff
@@ -1,4 +1,4 @@
-import type { Language } from '@logto/phrases';
+import { LanguageKey } from '@logto/shared';
```
Comment (gao-sun) [inline packages/connector-core/src/types.ts:1]: keep `import type`?

File: packages/console/src/pages/SignInExperience/types.ts [hunk 4]
```diff
@@ -1,5 +1,5 @@
-import { Language } from '@logto/phrases';
 import { SignInExperience, SignInMethodKey } from '@logto/schemas';
+import { LanguageKey } from '@logto/shared';
```
Comment (gao-sun) [inline packages/console/src/pages/SignInExperience/types.ts:2]: `import type`?

File: packages/ui/src/containers/SocialSignIn/SocialSignInDropdown/index.tsx [hunk 5]
```diff
@@ -48,7 +48,7 @@ const SocialSignInDropdown = ({ isOpen, onClose, connectors, anchorRef }: Props)
     >
       {connectors.map((connector) => {
         const { id, name, logo, logoDark } = connector;
-        const localName = isKeyOf(language, name) ? name[language] : name[Language.English];
+        const localName = name[languageKeyGuard.default('en').parse(language)];
```
Comment (gao-sun) [inline packages/ui/src/containers/SocialSignIn/SocialSignInDropdown/index.tsx:51]: 👍

**Review by gao-sun — APPROVED** [review]

File: packages/connector-core/src/types.ts (hunk 3 above)
Comment (wangsijie) [reply packages/connector-core/src/types.ts]: Updated (also other places)

File: packages/core/src/utils/zod.ts [hunk 6]
```diff
@@ -111,7 +112,7 @@ export const zodTypeToSwagger = (config: unknown): OpenAPIV3.SchemaObject => {
     };
   }
 
-  if (config instanceof ZodNativeEnum) {
+  if (config instanceof ZodNativeEnum || config instanceof ZodEnum) {
```
Comment (darcyYe) [inline packages/core/src/utils/zod.ts:115]: 👍

**Review by darcyYe — APPROVED** [review]
Cool

**Pull request merged**